waku/v2/node/peer_manager/waku_peer_store.nim

```diff
@@ -20,74 +25,107 @@ type
 
-proc get*(peerStore: WakuPeerStore,
+proc get*(peerStore: PeerStore,
           peerId: PeerID): StoredInfo =
   ## Get the stored information of a given peer.
-  
   StoredInfo(
+    # Taken from nim-libp2
     peerId: peerId,
-    addrs: peerStore.addressBook[peerId],
-    protos: peerStore.protoBook[peerId],
-    publicKey: peerStore.keyBook[peerId]
+    addrs: peerStore[AddressBook][peerId],
+    protos: peerStore[ProtoBook][peerId],
+    publicKey: peerStore[KeyBook][peerId],
+    agent: peerStore[AgentBook][peerId],
+    protoVersion: peerStore[ProtoVersionBook][peerId],
+
+    # Extended custom fields
+    connection: peerStore[ConnectionBook][peerId],
+    disconnectTime: peerStore[DisconnectBook][peerId],
+    origin: peerStore[SourceBook][peerId],
   )
 
-proc peers*(peerStore: WakuPeerStore): seq[StoredInfo] =
+proc peers*(peerStore: PeerStore): seq[StoredInfo] =
   ## Get all the stored information of every peer.
-  
-  let allKeys = concat(toSeq(keys(peerStore.addressBook.book)),
-                       toSeq(keys(peerStore.protoBook.book)),
-                       toSeq(keys(peerStore.keyBook.book))).toHashSet()
+  let allKeys = concat(toSeq(peerStore[AddressBook].book.keys()),
+                       toSeq(peerStore[ProtoBook].book.keys()),
+                       toSeq(peerStore[KeyBook].book.keys())).toHashSet()
 
   return allKeys.mapIt(peerStore.get(it))
+
+proc peers*(peerStore: PeerStore, proto: string): seq[StoredInfo] =
+  # Return the known info for all peers registered on the specified protocol
+  peerStore.peers.filterIt(it.protos.contains(proto))
+
+proc peers*(peerStore: PeerStore, protocolMatcher: Matcher): seq[StoredInfo] =
+  # Return the known info for all peers matching the provided protocolMatcher
+  peerStore.peers.filterIt(it.protos.anyIt(protocolMatcher(it)))
+
+proc toRemotePeerInfo*(storedInfo: StoredInfo): RemotePeerInfo =
+  RemotePeerInfo.init(peerId = storedInfo.peerId,
+                      addrs = toSeq(storedInfo.addrs),
+                      protocols = toSeq(storedInfo.protos))
+
+
+proc connectedness*(peerStore: PeerStore, peerId: PeerID): Connectedness =
+  # Return the connection state of the given, managed peer
+  # TODO: the PeerManager should keep and update local connectedness state for peers, redial on disconnect, etc.
+  # TODO: richer return than just bool, e.g. add enum "CanConnect", "CannotConnect", etc. based on recent connection attempts
+  return peerStore[ConnectionBook].book.getOrDefault(peerId, NotConnected)
+
+proc hasPeer*(peerStore: PeerStore, peerId: PeerID, proto: string): bool =
+  # Returns `true` if peer is included in manager for the specified protocol
+  # TODO: What if peer does not exist in the peerStore?
+  peerStore.get(peerId).protos.contains(proto)
+
+proc hasPeers*(peerStore: PeerStore, proto: string): bool =
+  # Returns `true` if the peerstore has any peer for the specified protocol
+  toSeq(peerStore[ProtoBook].book.values()).anyIt(it.anyIt(it == proto))
```

Comment:
These two methods differ just with an "s" and have completely different behavior. Please consider renaming one or merging both of the into a single one and allow multiple input types using the `when` statement.

Reply:
i was just moving this procs, but yes. notes.